browser/brave_origin/brave_origin_service_factory.cc

```diff
@@ -154,8 +155,10 @@ BraveOriginServiceFactory::~BraveOriginServiceFactory() = default;
 std::unique_ptr<KeyedService>
 BraveOriginServiceFactory::BuildServiceInstanceForBrowserContext(
     content::BrowserContext* context) const {
-  // Pass the profile ID here
-  return std::make_unique<BraveOriginService>("");
+  Profile* profile = Profile::FromBrowserContext(context);
+  std::string profile_id =
+      GetProfileId(profile->GetPath().BaseName().AsUTF8Unsafe());
+  return std::make_unique<BraveOriginService>(profile_id);
```

Comment:
feel free to ignore this comment since it only pertains to style, but you could consider passing the profile path and call `GetProfileID` in the constructor instead to slightly simplify the factory.